### PR TITLE
Update spirv.Source.

### PR DESCRIPTION
**Pull request by haonanya**
This aligns with https://github.com/KhronosGroup/SPIRV-LLVM-Translator/commit/a84bfcaf829995cc1ab4038149f90f98c2f14e02

### CODE REVIEW

File: clang/lib/CodeGen/CodeGenModule.cpp [hunk 1]
```diff
@@ -1358,14 +1358,14 @@ void CodeGenModule::Release() {
         TheModule.getOrInsertNamedMetadata("opencl.spir.version");
     SPIRVerMD->addOperand(llvm::MDNode::get(Ctx, SPIRVerElts));
     // We are trying to look like OpenCL C++ for SPIR-V translator.
```
Comment (bader) [inline clang/lib/CodeGen/CodeGenModule.cpp:1360]: The language mode is called "C++ for OpenCL".
```suggestion
    // We are trying to look like C++ for OpenCL for SPIR-V translator.
```

File: clang/lib/CodeGen/CodeGenModule.cpp [hunk 2]
```diff
@@ -1358,14 +1358,14 @@ void CodeGenModule::Release() {
...
-    // 4 - OpenCL_CPP, 100000 - OpenCL C++ version 1.0
+    // 6 - CPP_for_OpenCL, 100000 - OpenCL C++ version 1.0
```
Comment (bader) [inline clang/lib/CodeGen/CodeGenModule.cpp:1361]:
```suggestion
    // 6 - CPP_for_OpenCL, 100000 - C++ for OpenCL version 1.0
```

**Review by bader — COMMENTED** [review]
There is definitely some mess in here.
SPIR-V spec still mentions OpenCL C++ as one of the source languages for OpSource instruction - https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#Source_Language. Technically, SPIR-V translator should define some LLVM IR level interface to emit this value.

On the other hand, clang never had full support for OpenCL C++ and the spec for this language was removed from Khronos site.

It's not really important for SYCL which value to use for OpSource instruction - it has no semantic impact on the program. Switching from OpenCL C++ to C++ for OpenCL seems okay to me.

To avoid this type of problem, we can add SYCL language to SPIR-V spec. At the same time, I don't see too much value for users in have a separate OpSource for SYCL.

(cc @bashbaug)

**Comment by Naghasan** [comment]
>  It's not really important for SYCL which value to use for OpSource instruction - it has no semantic impact on the program. Switching from OpenCL C++ to C++ for OpenCL seems okay to me.

Arguably the patch should set the value to `7` for SYCL...

**Comment by haonanya** [comment]
> > It's not really important for SYCL which value to use for OpSource instruction - it has no semantic impact on the program. Switching from OpenCL C++ to C++ for OpenCL seems okay to me.
> 
> Arguably the patch should set the value to `7` for SYCL...

Yes, set to SYCL is right, currently spirv sets spirv.Source according to opencl language version. Set to 7 will drop spirv.Source metadata when translate spirv to llvm IR https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/main/lib/SPIRV/SPIRVReader.cpp#L4834,L4835.

**Comment by bader** [comment]
> > > It's not really important for SYCL which value to use for OpSource instruction - it has no semantic impact on the program. Switching from OpenCL C++ to C++ for OpenCL seems okay to me.
> > 
> > 
> > Arguably the patch should set the value to `7` for SYCL...
> 
> Yes, set to SYCL is right, but currently spirv sets spirv.Source according to opencl language version. Set to 7 will drop spirv.Source metadata when translate spirv to llvm IR https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/main/lib/SPIRV/SPIRVReader.cpp#L4834,L4835.

@intel/dpcpp-spirv-reviewers, is there a problem to add SYCL language support to the SPIRV-LLVM-Translator?

**Comment by haonanya** [comment]
ping @intel/dpcpp-spirv-reviewers, thanks very much.

**Comment by haonanya** [comment]
If spirv doesn't drop the SourceLanguageOpenCL_CPP metadata, the PR is not necessary.

**Pull request closed without merging**